core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -507,6 +520,27 @@ private boolean canProcessDictionary(Page page)
         return true;
     }
 
+    private boolean canProcessLowCardinalityDictionary(Page page)
+    {
+        // We don't have to rely on 'optimizer.dictionary-aggregations' here since there is little to none chance of regression
```

Comment:
nit: add extra empty line. Otherwise it looks like related to `if (channels.length <= 1) {`